### PR TITLE
otp changes

### PR DESCRIPTION
**Pull request by poonam20nikam**

### CODE REVIEW

File: src/controllers/User/GenerateOtp.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const Joi = require("@hapi/joi");
+const validator = require("express-joi-validation").createValidator({
+  passError: true,
+});
+
+const User = require("../../models/User");
+const { exist } = require("@hapi/joi");
```
Comment (vrishikesh) [inline src/controllers/User/GenerateOtp.js]: Remove this line it is not being used anywhere

File: src/controllers/User/GenerateOtp.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+});
+
+exports.validator = validator.body(schema);
+
+exports.handler = async (req, res, nex) => {    
+ const  userId    = req.body.userId; 
+ const  new_otp   = Math.floor(Math.random() * 1000000);
+
+  try {    
+    const userInsertData = User.findByIdAndUpdate(userId,
```
Comment (vrishikesh) [inline src/controllers/User/GenerateOtp.js]: Change this await call we are using await everywhere so lets be consistent

File: src/controllers/User/VerifyOtp.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+const Joi = require("@hapi/joi");
+const validator = require("express-joi-validation").createValidator({
+  passError: true,
+});
+
+const User = require("../../models/User");
+const { exist } = require("@hapi/joi");
```
Comment (vrishikesh) [inline src/controllers/User/VerifyOtp.js]: Not being used

File: src/controllers/User/VerifyOtp.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+exports.validator = validator.body(schema);
+
+exports.handler = async (req, res, nex) => {
+  const userId  = req.body.userId;
+  const otp     = req.body.otp;
+  const fieldsToSelect ="otp";
+  try {    
+    const user = await User.findById(userId).select(fieldsToSelect);
+    
+    if (user.otp == otp) {
```
Comment (vrishikesh) [inline src/controllers/User/VerifyOtp.js]: Otp should be an array and not a single value
So you need to loop thorugh all otps and check whether you have found it
If not found the message Should be Invalid OTP
If found you need to clear all otps against the user and then send reponse

File: src/models/User.js [hunk 5]
```diff
@@ -17,7 +17,7 @@ const userSchema = new Schema(
       },
     },
     otp: {
-      type: String,
+      type: [Number],
```
Comment (vrishikesh) [inline src/models/User.js]: Why?

File: src/routes/users.js [hunk 6]
```diff
@@ -288,4 +290,69 @@ router.post(
   registerUser.handler
 ); 
 
+/**
+ * @swagger
+ *
+ * /users/verifyOtp:
+ *  post:
+ *    description: Verifies whether the passed username is available or not
```
Comment (vrishikesh) [inline src/routes/users.js]: It verifies whether otp is correct or not

File: src/routes/users.js [hunk 7]
```diff
@@ -288,4 +290,69 @@ router.post(
...
+ *        type: string
+ * 
+ *      - name: otp
+ *        description: Enter otp check with user otp 
+ *        in: rawJson
+ *        required: true
+ *        type: number
+ *    responses:
+ *      200:
+ *        description: Returns string Available or Not Available
```
Comment (vrishikesh) [inline src/routes/users.js]: ?

File: src/controllers/User/GenerateOtp.js [hunk 8]
```diff
@@ -13,21 +11,21 @@ const schema = Joi.object().keys({
...
-            if(err){
-              res.fail("otp not send");
-            }else{
-              res.success("otp send", new_otp);
-            }
+    const  userId    = req.body.userId; 
+    const  new_otp   = Math.floor(100000 + Math.random() * 900000);
+  try {   
+    const user = await User.findById(userId);
+    if(user.otp.length+1 <=3){
```
Comment (vrishikesh) [inline src/controllers/User/GenerateOtp.js]: Remove this condition

Comment (vrishikesh) [reply src/controllers/User/GenerateOtp.js]: The condition is not going to be for otp sending it is going to be for otp attempt, also for that create global config MAX_OTP_ATTEMPT and check from there

File: src/controllers/User/VerifyOtp.js [hunk 9]
```diff
@@ -1,31 +1,31 @@
...
-  const otp     = req.body.otp;
-  const fieldsToSelect ="otp";
-  try {    
-    const user = await User.findById(userId).select(fieldsToSelect);
-    
-    if (user.otp == otp) {
-      res.success("verified", otp);
-    } else {
-      res.fail("otp not verified");
+      if ((user.length)>0) {
```
Comment (vrishikesh) [inline src/controllers/User/VerifyOtp.js]: Here add checking for max attempt and if max attempt is greater than configured value block this user id

File: src/models/User.js [hunk 10]
```diff
@@ -84,6 +84,14 @@ const userSchema = new Schema(
     interests: {
       type: [String],
     },
+    isLogin: {
```
Comment (vrishikesh) [inline src/models/User.js]: Why this flag when we already have isLoggedIn ?

File: src/models/User.js [hunk 11]
```diff
@@ -84,6 +84,14 @@ const userSchema = new Schema(
...
+      type: Number,
+      default: 0,
+    },
+    isRegister: {
```
Comment (vrishikesh) [inline src/models/User.js]: Same here

**Review by nikhilmishra1998 — APPROVED** [review]

**Pull request merged**